src/ui/components/Scanner/Scanner.tsx

```diff
@@ -68,26 +72,31 @@ const Scanner = forwardRef((props, ref) => {
         BarcodeScanner.hideBackground();
         const result = await startScan();
         if (result.hasContent) {
+          if (props.handleContent) props.handleContent(result.content);
           stopScan();
           // @TODO - foconnor: instead of setting the optype to idle we should
           // have a loading screen with "waiting for server..." etc,
           // and it can update to an error if the QR is invalid with a re-scan btn
-          dispatch(setCurrentOperation(OperationType.IDLE));
-          // @TODO - foconnor: when above loading screen in place, handle invalid QR code
-          await AriesAgent.agent.connections.receiveInvitationFromUrl(
-            result.content
-          );
+
+          if (currentRoute?.path !== RoutePath.OOBI_SCANNER) {
+            dispatch(setCurrentOperation(OperationType.IDLE));
+            // @TODO - foconnor: when above loading screen in place, handle invalid QR code
+            await AriesAgent.agent.connections.receiveInvitationFromUrl(
+              result.content
+            );
+          }
         }
       }
     }
   };
 
   useEffect(() => {
     if (
-      (currentRoute?.path === TabsRoutePath.SCAN ||
-        currentOperation === OperationType.SCAN_CONNECTION) &&
-      currentToastMsg !== ToastMsgType.CONNECTION_REQUEST_PENDING &&
-      currentToastMsg !== ToastMsgType.CREDENTIAL_REQUEST_PENDING
+      currentRoute?.path === TabsRoutePath.SCAN ||
+      currentRoute?.path === RoutePath.OOBI_SCANNER ||
+      (currentOperation === OperationType.SCAN_CONNECTION &&
```

Comment:
This logic is slightly different since the brackets moved - do all 3 scanners still work?

Reply:
Instead, now i keep it as it was but adding a extra condition just to let it enter the initScan if we are in oobi scanner:

`currentRoute?.path === RoutePath.OOBI_SCANNER ||`

```
useEffect(() => {
    if (
      currentRoute?.path === RoutePath.OOBI_SCANNER ||
      ((currentRoute?.path === TabsRoutePath.SCAN ||
        currentOperation === OperationType.SCAN_CONNECTION) &&
        currentToastMsg !== ToastMsgType.CONNECTION_REQUEST_PENDING &&
        currentToastMsg !== ToastMsgType.CREDENTIAL_REQUEST_PENDING)
    ) {
      initScan();
    } else {
      stopScan();
    }
  }, [currentOperation, currentRoute]);
```

In this way we dont modify the previous logic

Reply:
ok this works